diff --git a/include/zeep/el/object.hpp b/include/zeep/el/object.hpp
index 9d5b8e1..e1d5a7c 100644
--- a/include/zeep/el/object.hpp
+++ b/include/zeep/el/object.hpp
@@ -22,7 +22,7 @@
 #include <vector>
 #include <version>
 
-#if defined(__cpp_lib_flat_map)
+#if defined(__cpp_lib_flat_map) and not defined(__GLIBCXX__)
 # include <flat_map>
 #else
 # include <map>
@@ -130,7 +130,7 @@ class object
 
 	using nullptr_type = std::nullptr_t;
-#if defined(__cpp_lib_flat_map)
+#if defined(__cpp_lib_flat_map) and not defined(__GLIBCXX__)
 	using object_type = std::flat_map<std::string, object>;
 #else
 	using object_type = std::map<std::string, object>;
 #endif
